src/simplnx/Utilities/Parsing/DREAM3D/Dream3dIO.cpp

```diff
@@ -882,6 +892,38 @@ Result<IDataArray*> readLegacyDataArray(DataStructure& dataStructure, const nx::
     return MakeErrorResult<IDataArray*>(error.code, error.message);
   }
 
+  // Let's do some sanity checking of the tuple dims and the comp dims
+  // This is here because some 3rd party apps are trying to write .dream3d
+  // files but are getting confused about the Tuple Dims and the component
+  // dims. This bit of code will try to fix the specific issue where the
+  // .dream3d file essentially had _all_ the dimensions shoved in the
+  // tuple dimensions.
+  if(!amTDims.empty()) // If the amTDims are empty we are just reading a Data Array without an Attribute Matrix so skip.
```

Comment:
If we detect this issue, we should emit a warning.

Reply:
Done.